html/captive-portal/templates/saml.html

```diff
@@ -1,7 +1,14 @@
-<div class="c-card o-layout o-layout--center u-padding">
-  <div class="o-layout__item u-1/1 u-2/3@tablet u-3/5@desktop">
-    <a href="/saml/redirect" class="c-btn c-btn--primary u-1/1">
-      [% i18n_format("Click here to login on %s", source.description) %]
-    </a>
+<!-- [% source.type %] -->
+<form action="/saml/redirect" method="GET">
+
+  <div class="c-card u-padding u-padding-top-none">
+    [% form.get_field('aup').render | none %]
   </div>
-</div>
+      
+  <script src="/common/aup-autosubmit.js"></script>
+
+  <button id="button" type="submit" name="submit" class="c-btn c-btn--primary u-1/1 c-card--hidden">
+      [% i18n('Login') %]
+  </button>
```

Comment:
Unless I'm missing something, there is no need for this code here.

This code add an invisible button below "I accept the terms", you can click on it.

Reply:
That "I accept the terms" needs a target to submit the form

You can try to remove it and we'll see if it still works but AFAIK this is needed

Reply:
You're right.